tour_model_eval/load_predict.py

```diff
@@ -64,18 +81,28 @@ def predict_labels(trip):
             sel_fl = fl
             break
     if not sel_fl:
-        return {}
+        return []
     # choose selected model
     sel_model = models[sel_fl]
     # predict 2nd label for the new trip
+    # the value of sel_model.predict([trip_feat]) by default is numpy.ndarray, e.g.[1]
+    # so we need to turn it into '1' so that it can be the same type as dict key in user_labels dicts
     sel_sl = str(sel_model.predict([trip_feat])[0])
-    # values of the selected 1st round label
-    sel_1st_round_val = user_labels[sel_fl][0]
-    second_label_ls = list(sel_1st_round_val.keys())
+
+    # - seccond_round_result: values of the key of selected 1st round label
+    # e.g.{'0': [{'labels': {'mode_confirm': 'shared_ride', 'purpose_confirm': 'home', 'replaced_mode': 'drove_alone'},
+    # 'p': 1.0}],
+    # '1': [{'labels': {'mode_confirm': 'shared_ride', 'purpose_confirm': 'church', 'replaced_mode': 'drove_alone'},
+    # 'p': 0.9333333333333333},{'labels': {'mode_confirm': 'shared_ride', 'purpose_confirm': 'entertainment',
+    # 'replaced_mode': 'drove_alone'},'p': 0.06666666666666667}]}
+    # more explanation: '0' and '1' are 2nd round clusters from a bin(1st round cluster)
+    # cluster '0' contains all user label combinations and probabilities in this cluster
+    seccond_round_result = user_labels[sel_fl][0]
+    second_label_ls = list(seccond_round_result.keys())
     if sel_sl not in second_label_ls:
-        return {}
+        return []
```

Comment:
why do you need this? I think you can directly use `sel_ls not in second_round_result`

Reply:
Because Gabriel's function need a list, I follow the result from `placeholder_predictor_2(trip)`

Reply:
this is not about the return type. It is about the need for `second_label_ls = list(seccond_round_result.keys())`

Reply:
Oh ok, I see your point.